src/objects.c

```diff
@@ -1207,6 +1232,56 @@ bool evict_user_connection(PgUser *user)
 	return false;
 }
 
+/* evict the single oldest active connection from among all pools to make room in the user */
+static bool evict_active_user_connection(PgUser *user)
+{
+	struct List *item;
+	PgPool *pool;
+	PgSocket *oldest_connection = NULL;
+
+	statlist_for_each(item, &pool_list) {
+		pool = container_of(item, PgPool, head);
+		if (pool->user != user)
+			continue;
+		oldest_connection = compare_connections_by_time(oldest_connection, last_socket(&pool->active_server_list));
+	}
+
+	if (oldest_connection) {
+		disconnect_server(oldest_connection, true, "evicted");
+		return true;
+	}
+	return false;
+}
+
+static void enforce_user_connection_limit(evutil_socket_t sock, short flags, void *arg)
+{
+	PgUser *user = (PgUser*)arg;
+	int max = user_max_connections(user);
+	/* no limit to enforce if user is allocated unlimited connections */
+	if (max <= 0)
+		return;
+
+	// TODO: refers to total connection limit? test this!
+	while (user->connection_count > max) {
+		if (evict_idle_user_connection(user))
+			continue;
+		if (evict_active_user_connection(user))
+			continue;
+		/* user has no idle, unused or active connections to evict */
+		return;
+	}
+}
+
+void notify_user_cf_updated(PgUser *user)
+{
+	if (user->connection_count <= user_max_connections(user) || pgb_event_base == NULL)
+		return;
+
+	struct event ev_connection_limit_enforcement;
+	event_assign(&ev_connection_limit_enforcement, pgb_event_base, -1, EV_TIMEOUT, enforce_user_connection_limit, user);
```

Comment:
We queue a new event callback to drop all of the user's connections that surpass their newly configured limit, upon `RELOAD` or `SET USER ...`.